File: Dtos/VacancyDto.cs
using MtpApp.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace MtpApp.Dtos
{
    public class VacancyDto
    {
        public int Id { get; set; }

        [Required]
        [Display(Name = "Posting Date")]
        public DateTime PostingDate { get; set; }

        [Required]
        public DateTime Deadline { get; set; }

        [Required]
        [Display(Name = "Employer")]
        public int EmployerId { get; set; }
        public Employer Employers { get; set; }

        [Required]
        [Display(Name = "Job Position")]
        public int JobPositionId { get; set; }
        public JobPosition JobPositions { get; set; }

        [Required]
        [Display(Name = "Job Category")]
        public int JobCategoryId { get; set; }
        public JobCategory JobCategories { get; set; }

        [Required]
        [Display(Name = "Position Available")]
        public int PositionAvailable { get; set; }

        [Display(Name = "Contract Type")]
        [StringLength(255)]
        public string ContractType { get; set; }

        [StringLength(255)]
        public string Schedule { get; set; }

        [Required]
        public double Salary{ get; set; }

        [Required]
        public double Salarymax { get; set; }

        [StringLength(510)]
        public string Location { get; set; }

        public string Responsibilities { get; set; }

        public string Requirement { get; set; }

        public string ApplicationInformation { get; set; }

        [Required]
        public string Status { get; set; }

        public int ViewCount { get; set; }

        [StringLength(255)]
        public string Branch { get; set; }

        public VacancyDto()
        {
            ViewCount = ViewCount + 1;
        }
    }
}

File: spring-backend/src/main/java/com/mtp/api/dto/VacancyDto.java
package com.mtp.api.dto;

import lombok.Data;
import jakarta.validation.constraints.*;
import java.time.LocalDateTime;

@Data
public class VacancyDto {
    private Integer id;
    private LocalDateTime postingDate;
    private LocalDateTime deadline;
    private String closingDate;   // ISO date string "2024-01-01" from frontend

    @NotNull(message = "Employer is required")
    private Integer employerId;
    private String employerName;

    @NotNull(message = "Job position is required")
    private Integer jobPositionId;
    private String jobPositionName;

    private Integer jobCategoryId;
    private String jobCategoryName;

    @Min(value = 1, message = "At least one position must be available")
    private int positionAvailable;

    @NotBlank(message = "Contract type is required")
    private String contractType;

    private String salary;

    @NotBlank(message = "Status is required")
    private String status;
    private String imageUrl;
    
    // Additional UI fields
    private String responsibilities;
    private String requirement;
    private String applicationInformation;
    private String schedule;
    private String location;
    private String salarymax;
}
